Makefile

```diff
@@ -178,6 +185,13 @@ C_FILES := $(foreach dir,$(SRC_DIRS),$(wildcard $(dir)*.c))
 # Object files
 O_FILES := 	$(foreach file,$(S_FILES),$(BUILD_DIR)/$(file:.s=.o)) \
             $(foreach file,$(C_FILES),$(BUILD_DIR)/$(file:.c=.o))
+			
+# Automatic dependency files
+DEP_FILES := $(O_FILES:.o=.d) $(ULTRA_O_FILES:.o=.d) $(GODDARD_O_FILES:.o=.d) $(BUILD_DIR)/$(LD_SCRIPT).d
+			
+# Check code syntax with host compiler
+CC_CHECK := gcc
+CC_CHECK_CFLAGS := -fsyntax-only -fsigned-char $(CC_CFLAGS) $(INCLUDE_CFLAGS) $(DEF_INC_CFLAGS) -std=gnu90 -Wall -Wextra -Wno-format-security -Wno-main -DNON_MATCHING -DAVOID_UB -m32 -D_LANGUAGE_C
```

Comment:
What's the `-m32` option for? That option isn't supported on my ARM box, but removing it doesn't seem to have an impact I can notice.

Reply:
It's a flag that makes sure GCC treats the code as 32 bit. It was actually behind a check and I removed it, so I can add it back in later to test.